compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1627,9 +1574,40 @@ impl<'a> Parser<'a> {
         if self.eat(&token::Semi) {
             return Ok(());
         }
+
+        if self.recover_colon_as_semi() {
+            return Ok(());
```

Comment:
Adding a note to all the new places where we recover to make sure that we don't regress anything.
Here the recovery is okay because we failed the `eat` above and are on an error path ✅